module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/RenewalsHelperTest.php

```diff
@@ -0,0 +1,433 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFindTest\ILS\Logic;
+
+use Laminas\Mvc\Plugin\FlashMessenger\FlashMessenger;
+use Laminas\Stdlib\Parameters;
+use PHPUnit\Framework\TestCase;
+use VuFind\ILS\Connection;
+use VuFind\ILS\Logic\RenewalsHelper;
+use VuFind\Validator\CsrfInterface;
+
+use function in_array;
+use function is_array;
+
+/**
+ * Unit tests for the RenewalsHelper
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class RenewalsHelperTest extends TestCase
+{
+    /**
+     * Test adding renew details when using an external link.
+     *
+     * @return void
+     */
+    public function testAddRenewDetailsWithLink(): void
+    {
+        $catalog = $this->getMockBuilder(Connection::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['__call'])
+            ->getMock();
+
+        $ilsDetails = ['id' => 'item123', 'title' => 'Test Item'];
+        $renewStatus = ['function' => 'renewMyItemsLink'];
+        $expectedLink = '/MyResearch/renew/item123';
+
+        $catalog->expects($this->once())
+            ->method('__call')
+            ->with(
+                $this->equalTo('renewMyItemsLink'),
+                $this->equalTo([$ilsDetails])
+            )
+            ->willReturn($expectedLink);
+
+        $helper = new RenewalsHelper();
+        $result = $helper->addRenewDetails($catalog, $ilsDetails, $renewStatus);
+
+        $this->assertArrayHasKey('renew_link', $result);
+        $this->assertEquals($expectedLink, $result['renew_link']);
+        $this->assertArrayNotHasKey('renew_details', $result);
+    }
+
+    /**
+     * Test adding renew details when using a form.
+     *
+     * @return void
+     */
+    public function testAddRenewDetailsWithForm(): void
+    {
+        $catalog = $this->getMockBuilder(Connection::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['__call'])
+            ->getMock();
+
+        $ilsDetails = ['id' => 'item456', 'title' => 'Another Item'];
+        $renewStatus = ['function' => 'renewMyItems'];
+        $expectedDetails = ['form_data' => 'details_here'];
+
+        $catalog->expects($this->once())
+            ->method('__call')
+            ->with(
+                $this->equalTo('getRenewDetails'),
+                $this->equalTo([$ilsDetails])
+            )
```

Comment:
If I'm not mistaken above, then this should also work:

```suggestion
            ->with('getRenewDetails', [$ilsDetails])
```

(And there are more instances of `$this->equalTo` below that might be able to be simplified too, if you agree that it improves readability).